bark/world/evaluation/evaluator_rss.hpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2019 fortiss GmbH, Julian Bernhard, Klemens Esterle, Patrick
+// Hart, Tobias Kessler
+//
+// This work is licensed under the terms of the MIT license.
+// For a copy, see <https://opensource.org/licenses/MIT>.
+
+#ifndef BARK_WORLD_EVALUATION_EVALUATOR_RSS_HPP_
+#define BARK_WORLD_EVALUATION_EVALUATOR_RSS_HPP_
+
+#include <limits>
+#include <memory>
+#include <string>
+
+#include "bark/world/evaluation/base_evaluator.hpp"
+#include "bark/world/evaluation/rss_interface.hpp"
+#include "bark/world/observed_world.hpp"
+#include "bark/world/world.hpp"
+
+namespace bark {
+namespace world {
+namespace evaluation {
+
+class EvaluatorRss : public BaseEvaluator {
+ public:
+  EvaluatorRss() : agent_id_(std::numeric_limits<AgentId>::max()) {}
+
+  explicit EvaluatorRss(
+      const AgentId& agent_id, const std::string& opendrive_file_name,
+      const std::vector<float>& default_vehicle_dynamics =
+          std::vector<float>{3.5, -8., -4., -3., 0.2, -0.8, 0.1, 1.},
+      const std::unordered_map<AgentId, std::vector<float>>&
+          agents_vehicle_dynamics =
+              std::unordered_map<AgentId, std::vector<float>>(),
+      const float& discretize_step = 1.,
+      const float& checking_relevent_range = 1.,
+      const float& route_predict_range = 50.)
+      : agent_id_(agent_id),
+        rss_(opendrive_file_name, default_vehicle_dynamics,
+             agents_vehicle_dynamics, discretize_step, checking_relevent_range,
+             route_predict_range) {}
+
+  virtual ~EvaluatorRss() {}
+
+  // Returns a boolean indicating the safety response of the specified agent.
+  // True if for each nearby agents, at least one of the all possible RSS
+  // situations is safe, false if unsafe, uninitialized (none in python) if no
+  // Rss check can be performed.
+  virtual EvaluationReturn Evaluate(const World& world) {
+    return rss_.GetSafetyReponse(world, agent_id_);
+  };
+
+  // Returns an unorder_map indicating the pairwise safety respone of the
+  // specified agent to every other nearby agents. Key is AgentId of an nearby
```

Comment:
must be true if all, right?

Reply:
I was following the logic in the rss library [here](https://github.com/intel/ad-rss-lib/blob/ffa69395dcce65dc16b8a3eb808541e04e3b22bc/ad_rss/impl/include/ad/rss/state/RssStateOperation.hpp#L80), which considers a state as safe if one of the directional situation is safe.

In the paper, I find this sentence is relevant to it: `In order to have a collision between two cars, they must be both at a non-safe longitudinal distance and at a non-safelateral distance.`

I think it is open to interpret by us, I can change it to AND relation instead of OR.

Reply:
I think this is the correct behavior. For example, in RSS, if a vehicle is following another vehicle at the back on the same lane, the longitudinal situation will be safe but the latitudinal will not. What do you think? Is this OR behavior is okay? @patrickhart